File: java/src/main/java/com/example/gemini/SystemInstruction.java
/*
 * Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.example.gemini;

import com.google.common.collect.ImmutableList;
import com.google.genai.Client;
import com.google.genai.types.Content;
import com.google.genai.types.GenerateContentConfig;
import com.google.genai.types.GenerateContentResponse;
import com.google.genai.types.Part;
import org.jspecify.annotations.Nullable;

public class SystemInstruction {
    public static @Nullable String systemInstruction() {
        // [START system_instruction]
        Client client = new Client();

        Part textPart = Part.builder().text("You are a cat. Your name is Neko.").build();

        Content content = Content.builder().role("system").parts(ImmutableList.of(textPart)).build();

        GenerateContentConfig config = GenerateContentConfig.builder()
                .systemInstruction(content)
                .build();

        GenerateContentResponse response =
                client.models.generateContent(
                        "gemini-3.8-flash",
                        "Good morning! How are you?",
                        config);

        System.out.println(response.text());
        // [END system_instruction]
        return response.text();
    }
}

File: go/system_instruction.go
package examples

import (
	"context"
	"os"
	"log"

	"google.golang.org/genai"
)

func SystemInstruction() error {
	// [START system_instruction]
	ctx := context.Background()
	client, err := genai.NewClient(ctx, &genai.ClientConfig{
		APIKey:  os.Getenv("GEMINI_API_KEY"),
		Backend: genai.BackendGeminiAPI,
	})
	if err != nil {
		log.Fatal(err)
	}

	// Construct the user message contents.
	contents := []*genai.Content{
		genai.NewContentFromText("Good morning! How are you?", genai.RoleUser),
	}

	// Set the system instruction as a *genai.Content.
	config := &genai.GenerateContentConfig{
		SystemInstruction: genai.NewContentFromText("You are a cat. Your name is Neko.", genai.RoleUser),
	}

	response, err := client.Models.GenerateContent(ctx, "gemini-3.8-flash", contents, config)
	if err != nil {
		log.Fatal(err)
	}
	printResponse(response)
	// [END system_instruction]
	return err
}
